src/duplex.ts

```diff
@@ -217,9 +219,12 @@ function _generate(mirror, obj, patches, path) {
         }
       }
     }
-    else {
+    else if(Array.isArray(mirror) === Array.isArray(obj)) {
```

Comment:
Could you add some code comments about what's happening here? Why for two arrays or two non-arrays it's a `delete`, but for array and non-array, it's a `replace`?

Reply:
I'll add something of the effect of: Because the `replace` is for the whole obj, not for `obj[key]`.